src/signMsgAuthToken.c

```diff
@@ -0,0 +1,213 @@
+#include "getPrivateKey.h"
+#include "signMsgAuthToken.h"
+#include "addressHelpers.h"
+#include "utils.h"
+
+typedef struct {
+    char address[62];
+    uint32_t len;
+    uint8_t hash[32];
+    uint8_t signature[64];
+    char token[MAX_DISPLAY_DATA_SIZE];
+} token_auth_context_t;
+
+static token_auth_context_t token_auth_context;
+
+static uint8_t set_result_auth_token();
+bool sign_auth_token(void);
+
+// UI for confirming the message hash on screen
+UX_STEP_NOCB(
+    ux_auth_token_msg_flow_33_step, 
+    bnnn_paging, 
+    {
+      .title = "Address",
+      .text = token_auth_context.address,
+    });
+UX_STEP_NOCB(
+    ux_auth_token_msg_flow_34_step, 
+    bnnn_paging, 
+    {
+      .title = "Auth Token",
+      .text = token_auth_context.token,
+    });    
+UX_STEP_VALID(
+    ux_auth_token_msg_flow_35_step, 
+    pb, 
+    send_response(set_result_auth_token(), true),
+    {
+      &C_icon_validate_14,
+      "Authorize",
+    });
+UX_STEP_VALID(
+    ux_auth_token_msg_flow_36_step, 
+    pb,
+    send_response(0, false),
+    {
+      &C_icon_crossmark,
+      "Reject",
+    });
+
+UX_FLOW(ux_auth_token_msg_flow,
+  &ux_auth_token_msg_flow_33_step,
+  &ux_auth_token_msg_flow_34_step,
+  &ux_auth_token_msg_flow_35_step,
+  &ux_auth_token_msg_flow_36_step
+);
+
+void init_auth_token_context(void) {
+    bip32_account = 0;
+    bip32_address_index = 0;
+
+    app_state = APP_STATE_IDLE;
+}
+
+void update_token_display_data(uint8_t *data_buffer, uint8_t data_length) {
```

Comment:
Usually, when setting function parameters (especially mixed input/output) you specify their intent by using const, e.g. here they are only used as input, not modified:

```suggestion
void update_token_display_data(uint8_t const *data_buffer, uint8_t const data_length) {
```

Reply:
done